Macros/hmda_tracts_new.sas

```diff
@@ -0,0 +1,273 @@
+/**************************************************************************
+ Program:  hmda_tracts_new.sas
+ Library:  HMDA
+ Project:  Urban-Greater DC
+ Author:   Rob Pitingolo
+ Created:  12/6/18
+ Version:  SAS 9.4
+ Environment:  Windows 7
+ 
+ Description:  Macro to create tract-level HMDA files from raw HMDA.
+ 
+ Modifications:
+ 
+**************************************************************************/
+
+%macro hmda_tracts_new (start_yr,end_yr);
+
+%do year = &start_yr. %to &end_yr.;
+
+%if %upcase( &year ) < 2012 %then %do;
+	%let geo = geo2000;
+%end;
+
+%else %if %upcase( &year ) >= 2012 %then %do;
+	%let geo = geo2010;
+%end;
+
+data apps&year.;
+	set hmda.loans_&year.;
+
+	/*Keep DC records */
+	if ucounty = "11001";
+
+	/* Format and keep valid formats */
+	format &geo. geo00a.;
+
+	%let geo = %upcase( &geo. );
+    %let geovfmt = %sysfunc( putc( &geo, $geovfmt. ) );
+
+    if put( &geo., &geovfmt ) ^= " ";
+
+
+	/* Flag each record as an application */
+	app=1;
+
+	/* Race flags */
+	%racecalc_04;
+
+	/* Gender flags */
+	%gender;
+
+	/* Purpose type flags */
+	%purp_type;
+
+	/* Property type flags */
+	%prop_type;
+
+	/* Income recodes */
+	if income < 0 then hhincome = .;
+		else hhincome=income;
+	if amount < 0 then amount = .;
+		else amount=amount;
+
+	/* Loan type flags */
+	if loantype = '1' then conv = 1;
+		else if loantype in ('2', '3' , '4') then govt=1;
+
+	/* Analysis flags */
+	length f_denial highFlag f_lien 3.;
+
+	if appdate ne '1' then do;
+		*** flag variable for HOEPA vars ****;
+		if hoepa = 1 then f_hoepa=1;
+			else f_hoepa=0;
+		if lien = '1' then f_lien = 1;
+			else if lien = '2' then f_lien = 0;
+	end;
+
+	else do;
+		f_lien  = . ;
+		f_hoepa = . ;
+			end;
+
+	/* Denials flags */
+	if action = '3' then f_denial =1;
+		else f_denial = 0;
+	
+	/* High interest flag */
+	if high_interest = "1" then highflag=1;
+		else highflag=0;
+
+	/* Create income categories from ipums %hud_inc macro */
+	if hhincome ^= . then do;
+	numprec = 4;
+	%Hud_inc_2009;
```

Comment:
It does not seem like we should always use the 2009 income macro. We should use the macro that corresponds to the HMDA data year.